Chess/ViewController.swift

```diff
@@ -97,6 +98,41 @@ class ViewController: UIViewController {
         })
         setSelection(nil)
     }
+*/
+
+/*
+Key Changes:
+
+Weak Capture in Animations: I added [weak self] to the animations block as well. This prevents a strong reference cycle in case the animation takes a long time or the view controller is dismissed before the animation completes.
```

Comment:
This is a misunderstanding of the animation life cycle. The `animations` block is executed immediately when the function is called, so there is no reason to use weak inside this block. The `completion` block is deferred until the end of the animation, so weak makes more sense there.